keps/77-dynamically-sized-jobs/README.md

```diff
@@ -7,168 +7,290 @@
+Once a WorkloadSlice is admitted, it will replace the original/old WorkloadSlice. 
+The latter will be deactivated and marked as finished.
+- Each WorkloadSlice inherits the resource flavor assignment of the original admitted slice, enforcing flavor consistency across scale-ups. 
 - In MultiKueue, Workload Slices would go into both clusters (management and workload) in a multi-cluster environment.
-- Workload Slices will be created by Kueue and use identical PodTemplates (which is already enforced by Kuberay in the case for RayCluster).
-- Workload Slices will belong to the same resource flavor as the top-level Workload that was initially admitted.
+- Workload Slices will be created by Kueue and use identical PodTemplate.
+- Workload Slices will belong to the same resource flavor as the top-level Workload that was initially admitted (for more details see "Resource Flavors" subsection)
+
+### Enablement
 
-The parent Workload should have a condition that reflects the scaling progression status. 
+WorkloadSlices in Kueue are enabled through a combination of a Kubernetes feature gate and an opt-in annotation on individual Workload objects. 
+At the cluster level, the WorkloadSlices feature must be enabled via the corresponding Kubernetes feature gate, which controls whether the controller logic for slicing is active. 
+
+Once the feature gate is enabled, individual Workload objects can opt into slicing by including the `kueue.x-k8s.io/elastic-job: "true"` annotation. 
+When both conditions are met, Kueue treats the Workload as eligible for partitioning into one or more WorkloadSlice objects, enabling fine-grained scheduling and 
+execution across multiple clusters or within a single cluster. If the feature gate is disabled or the annotation is omitted (or set to "false"), 
+the system defaults to the traditional single-Workload scheduling path for full backward compatibility.
+
+#### Features
+```golang
+// owner: @ichekrygin
+// kep: https://github.com/kubernetes-sigs/kueue/tree/main/keps/77-dynamically-sized-jobs
+//
+// ElasticJobsViaWorkloadSlices enables support for horizontal-scale in jobs.
+ElasticJobsViaWorkloadSlices featuregate.Feature = "ElasticJobsViaWorkloadSlices"
+```
 
+#### WorkloadSliceAnnotation
 ```golang
+// Workload slicing can be enabled for a specific integration job instance,
+// provided that the integration supports workload slicing.
 const (
-  ...
-  ...
-	// WorkloadResizeRequested means that the Workload is in the process of scaling up or down
-	WorkloadResizeRequested = "ResizeRequested"
+  // EnabledAnnotationKey refers to the annotation key present on Job's that support
+  // workload slicing.
+  // This annotation is alpha-level.
+  EnabledAnnotationKey = "kueue.x-k8s.io/elastic-job"
+  // EnabledAnnotationValue refers to the annotation value. To enable
+  // workload slicing for a given job, we match both annotation key and value.
+  EnabledAnnotationValue = "true"
```

Comment:
If this is just for backward compatibility, we probably can abandon this annotation when we promote the Workload API version to v1beta2.

But, that is not now.

Reply:
> the mechanism is based on scheduling gates, and this would mean performance hit for users not using ElasticJobs. I think removing the scheduling gate from every Pod can be felt by some large-scale users. Sure, for most users it will not be an issue, so we may have "AssumeAllJobsAreElastic" in global configuration, but I would defer it for later.

Actually, as my initial thought was moving this annotation functionality to Global Config (AssumeAllJobsAreElastic).
However, I wanted to think more if the config is useful. In other words, for now, this should be annotation. But, in the future versioin, it might not be annotation.